src/test/java/org/tarantool/ClientReconnectClusterIT.java

```diff
@@ -2,8 +2,9 @@
 
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
```

Comment:
As I see (correct me if I'm wrong) we have no integration test that checks we really fetch and use a new list of nodes after a user-provided delay. All of them test only an initial fetch. I think it worth to test it with some small timeout. Maybe it worth to write a Lua function, that will return different list of instances for each call, to check we really use a new list after a delay.

Reply:
Added a test ClientReconnectClusterIT.testDelayFunctionResultFetch that covers the case of yours.